osfoffline/filesystem_manager/osf_event_handler.py

```diff
@@ -173,21 +205,29 @@ def on_modified(self, event):
 
         if isinstance(event, DirModifiedEvent):
             return
-        src_path = ProperPath(event.src_path, event.is_directory)
-
-        # get item
         try:
-            item = self._get_item_by_path(src_path)
-        except ItemNotInDB:
-            # todo: create file folder
-            logging.warning('file {} was modified but not already in db. create it in db.'.format(src_path))
-            return  # todo: remove this once above is implemented
+            src_path = ProperPath(event.src_path, event.is_directory)
+        except Exception:
+            logging.exception('Exception caught: Invalid path')
+            AlertHandler.warn('invalid path specified. {} will not be synced'.format(event.src_path.split('/')[-1]))
+        else:
+            # get item
+            try:
+                item = self._get_item_by_path(src_path)
+            except ItemNotInDB:
+                # todo: create file folder
+                logging.warning('file {} was modified but not already in db. create it in db.'.format(src_path))
+                return  # todo: remove this once above is implemented
 
-        # update hash
-        item.update_hash()
+            # update hash
+            item.update_hash()
```

Comment:
This function actually does open a function and try to update hash. It may fail if the file is for some reason inaccessible; may be worth catching an `OsError` here.